Dockerfile

```
@@ -35,12 +35,9 @@ USER ${NB_UID}
 ENV CARGO_HOME="${HOME}/.cargo"
 ENV PATH="${CARGO_HOME}/bin:${PATH}"
 
-# Use solution from https://github.com/NASA-Openscapes/corn/blob/main/ci/Dockerfile
-# for installing VS Code extensions.
-COPY install-vscode-ext.sh ${HOME}/.kernels/install-vscode-ext.sh
-
-RUN bash ${HOME}/.kernels/install-vscode-ext.sh
-
-# Install icechunk development packages (these steps required from https://github.com/earth-mover/icechunk/issues/197)
-RUN python -m pip install xarray git+https://github.com/zarr-developers/VirtualiZarr.git@ab/upgrade-icechunk#egg=VirtualiZarr[icechunk]
-RUN python -m pip install git+https://github.com/mpiannucci/kerchunk@v3
+# Install from main branches of key libraries
+RUN python -m pip install git+https://github.com/pydata/xarray.git
+RUN python -m pip install git+https://github.com/zarr-developers/zarr-python.git
+RUN python -m pip install git+https://github.com/zarr-developers/VirtualiZarr.git
+RUN python -m pip install git+https://github.com/fsspec/kerchunk.git
+RUN python -m pip install git+https://github.com/earth-mover/icechunk#subdirectory=icechunk-python
```

Unless there's some compelling reason to use separate RUN commands, combine them into one to reduce the number of image layers and reduce the image size, which is also kept smaller by using the `--no-cache-dir` option to `pip`.

```suggestion
RUN python -m pip install --no-cache-dir \
    git+https://github.com/pydata/xarray.git \
    git+https://github.com/zarr-developers/zarr-python.git \
    git+https://github.com/zarr-developers/VirtualiZarr.git \
    git+https://github.com/fsspec/kerchunk.git \
    git+https://github.com/earth-mover/icechunk#subdirectory=icechunk-python
```